test/suites-ostree/staged-failed-upgrade.robot

```diff
@@ -0,0 +1,59 @@
+*** Settings ***
+Documentation       Tests related to upgrading MicroShift
+
+Resource            ../resources/common.resource
+Resource            ../resources/ostree.resource
+Library             Collections
+
+Suite Setup         Setup
+Suite Teardown      Teardown
+
+Test Tags           ostree
+
+
+*** Variables ***
+${USHIFT_HOST}      ${EMPTY}
+${USHIFT_USER}      ${EMPTY}
+
+${TARGET_REF}       ${EMPTY}
+
+
+*** Test Cases ***
+First Failed Staged Upgrade Should Restore Backup
+    [Documentation]    Performs an upgrade, when first unhealthy a restore is performend
+
+    Wait For Healthy System
+
+    ${initial_backup}=    Get Future Backup Name For Current Boot
+
+    TestAgent.Add Action For Next Deployment    1    fail_greenboot
+    Deploy Commit Not Expecting A Rollback    ${TARGET_REF}
```

Comment:
Why don't we expect a rollback here?

Reply:
In this situation we're verifying that if the first deployment is set to unhealthy just once, we expect the restore to be triggered, but after the first boot, things should roll as intended with out rollback. However, I think I might have a mis-conception here, added another validation to make sure the log statements are printing but I'm not seeing the restore logs.

Reply:
Ah, take a look at:
```
Deploy Commit Expecting A Rollback
    [Documentation]    Deploys given ref and configures test agent for failing greenboot.
    ...    It expects the system to roll back.
    [Arguments]    ${ref}    ${write_agent_cfg}=${TRUE}

    ${initial_deploy_id}=    Get Booted Deployment Id
    ${deploy_id}=    Rebase System    ${ref}
    IF    ${write_agent_cfg}    TestAgent.Write
```

`Deploy Commit Not Expecting A Rollback` does not persist the TestAgent's config, so the test didn't create the scenario we wanted - after staging and booting into a new deployment, there should be one auto-reboot, so on 2nd boot MicroShift should restore the backup of the first (rollback) deploy

Reply:
I implemented following change
```
diff --git a/test/resources/ostree.resource b/test/resources/ostree.resource
index 0efdc4b9b..d1b8ebeec 100644
--- a/test/resources/ostree.resource
+++ b/test/resources/ostree.resource
@@ -41,10 +41,11 @@ Deploy Commit Expecting A Rollback
 Deploy Commit Not Expecting A Rollback
     [Documentation]    Deploys given ref and configures test agent for failing greenboot.
     ...    It expects the system to roll back.
-    [Arguments]    ${target_ref}
+    [Arguments]    ${target_ref}     ${write_agent_cfg}=${FALSE}    
 
     ${initial_deploy_id}=    Get Booted Deployment Id
     ${deploy_id}=    Rebase System    ${target_ref}
+    IF    ${write_agent_cfg}    TestAgent.Write
     Reboot MicroShift Host
 
     Log To Console    "System rebased to ${target_ref} and rebooted - starting health checking"
diff --git a/test/suites-ostree/staged-failed-upgrade.robot b/test/suites-ostree/staged-failed-upgrade.robot
index 3f6ba1085..3b44bb3df 100644
--- a/test/suites-ostree/staged-failed-upgrade.robot
+++ b/test/suites-ostree/staged-failed-upgrade.robot
@@ -27,7 +27,7 @@ First Failed Staged Upgrade Should Restore Backup
     ${initial_backup}=    Get Future Backup Name For Current Boot
 
     TestAgent.Add Action For Next Deployment    1    fail_greenboot
-    Deploy Commit Not Expecting A Rollback    ${TARGET_REF}
+    Deploy Commit Not Expecting A Rollback    ${TARGET_REF}    ${TRUE}
 
     Wait For Healthy System
```

but I see that `test-agent` doesn't get to remove the artifacts:
```
Aug 03 03:12:00 rhel-9-2-microshift-source-failed-staged-host1 bash[771]: + sleep infinity
Aug 03 03:14:42 rhel-9-2-microshift-source-failed-staged-host1 bash[771]: Terminated
Aug 03 03:14:42 rhel-9-2-microshift-source-failed-staged-host1 bash[771]: ++ _cleanup
Aug 03 03:14:42 rhel-9-2-microshift-source-failed-staged-host1 bash[771]: ++ for cmd in "${CLEANUP_CMDS[@]}"
Aug 03 03:14:42 rhel-9-2-microshift-source-failed-staged-host1 bash[771]: ++ rm -v /etc/greenboot/check/required.d/99_microshift_test_failure.sh
Aug 03 03:14:42 rhel-9-2-microshift-source-failed-staged-host1 systemd[1]: Stopping MicroShift Test Agent...
Aug 03 03:14:42 rhel-9-2-microshift-source-failed-staged-host1 bash[771]: Terminated
```

but it should be (`-v` to `rm`):
```
+ sleep infinity
^C++ _cleanup
++ for cmd in "${CLEANUP_CMDS[@]}"
++ rm -v /etc/greenboot/check/required.d/99_microshift_test_failure.sh
removed '/etc/greenboot/check/required.d/99_microshift_test_failure.sh'
```

Reply:
After adding `exit 0` to `_cleanup()` in `microshift-test-agent.sh` it seems that the file gets deleted properly.
Spoke to Egli and he'll include that as well